tests/storage/fsp/test_fsp_sr.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+import os
+import logging
+from lib.common import wait_for, vm_image
+from lib.vdi import VDI
+
+DIRECTORIES_PATH = 'directories'
+
+class TestFSPSRCreateDestroy:
+    """
+    Tests that do not use fixtures that setup the SR
+    because they precisely need to test SR creation and destruction.
+    """
+
+    def test_create_and_destroy_sr(self, host):
+        db_path = host.ssh(['mktemp', '-d'])
+        host.ssh(['mkdir', db_path + '/' + DIRECTORIES_PATH])
+        sr = host.sr_create('fsp', "fsp-local-SR-test", {'file-uri': db_path})
+        sr.destroy()
+        host.ssh(['rm', '-rf', db_path])
+
+    def test_create_and_destroy_sr_non_existing_path(self, host):
+        # create and then remove a temp dir to
+        # get an unique non-existing path
+        db_path = host.ssh(['mktemp', '-d'])
+        host.ssh(['rm', '-rf', db_path])
```

Comment:
`mktemp -d --dry-run` would do the same with one less SSH call. The man page says it's unsafe, but I don't think it's less safe than creating then deleting the directory. In both cases the risk would be a name collision, which is very unlikely in the context of this test.